flink-core/src/main/java/org/apache/flink/util/FileLock.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.channels.OverlappingFileLockException;
+import java.nio.file.Paths;
+
+/** A file lock used for avoiding race condition among multiple unit test processes */
+public class FileLock {
+    private static final long TRY_LOCK_INTERVAL = 50;
+    private final File file;
+    private FileOutputStream outputStream;
+    private java.nio.channels.FileLock lock;
+
+    public FileLock(String fileName) {
+        this.file = new File(fileName);
```

Comment:
Here we should simply pass `file` directly and just perform a `checkNotNull`.

We should then add a factory method
```
static FileLock inTempFolder(String fileName) {
   return new File(System.getProperty("java.io.tmpdir"), fileName);
}
```

Maybe we should cache `System.getProperty("java.io.tmpdir")` in some constant `FileLock#TEMP_FOLDER`.

Lastly, do we need to strip `fileName` of illegal characters? I think `$` is a valid java class part but will lead to issues on some platforms.

Reply:
Good idea. I ignored the importance of Unix/Windows file name validation. I will add it.

Reply:
I added a normalization method which only allows letters, slashes and backslashes.